src/components/menu/menu.types.ts

```diff
@@ -57,4 +57,28 @@ export interface MenuItem<T = any> {
      * Value of the menu item.
      */
     value?: T;
+
+    /**
+     * What sub items the item contains
+     * Don't set if using lazy loading.
+     */
+    subItems?: Array<MenuItem<T>>;
```

Comment:
How about we make this something like this
```suggestion
    items?: Array<MenuItem<T>> | () => Promise<Array<MenuItem<T>>>;
```

This way, the consumer can either just send in a list of items for the nested menu if they are available, or give us a function to call that will load the items asynchronously when needed. This will allow us to get rid of both the `loadSubItems`, `lazyLoadItems` and `loading` props. The function will probably need some kind of input argument, but I don't know what that is yet :smile: Perhaps the parent item? :thinking:

Reply:
Maybe to root `items` prop also should have this type to allow lazy loading of the first level of items, not sure if it will make the code easier or harder to implement :smile:

Reply:
Implemented this solution and removed unnecessary props